NO-JIRA Fix CppCheck warnings

jdanekrh opened the pull request.

jdanekrh commented on proton-c/src/proactor/libuv.c:
```diff
@@ -1179,7 +1179,10 @@ pn_proactor_t *pn_proactor() {
   p->batch.next_event = &proactor_batch_next;
-  if (!p->collector) return NULL;
+  if (!p->collector) {
+    free(p);
```
I hope this is enough to free it at this point.

astitcher replied:
Looks good to me, though stylistically I'd move line 1181 after the 'if' statement.

jdanekrh replied:
I thought of that, but then decided to keep changes to a minimum. Will you move it or should I resubmit the PR? I'd be OK if you just committed it.

jdanekrh commented:
The Travis passed on my branch, https://travis-ci.org/jdanekrh/qpid-proton/builds/271996339

astitcher commented:
Do you need me to merge these changes? (I forget if you are a committer or not)

jdanekrh commented:
Yeah, I need a committer to review and merge it for me.

On Tue, Sep 5, 2017 at 6:19 PM, Andrew Stitcher <notifications@github.com>
wrote:

> Do you need me to merge these changes? (I forget if you are a committer or
> not)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/qpid-proton/pull/119#issuecomment-327227718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARDKn6YQHvSHpnqCtwvxO4bEE8nfkh-oks5sfXR9gaJpZM4PMvJl>
> .
>



-- 
Jiří Daněk
Messaging QA

michaelandrepearce commented:
@jdanekrh which version of cppcheck you using? using 1.8 we get a few more warnings, e.g. detects header file method param names order don't match cpp implementation.

e.g.
[proton-c/src/proactor/proactor-internal.h:49] -> [proton-c/src/proactor/proactor-internal.c:79]: (warning) Function 'pni_proactor_set_cond' argument order different: declaration 'cond, what, msg, host, port' definition 'cond, what, host, port, msg'

astitcher commented:
@jdanekrh ok - I'll merge and commit this - also note we need a JIRA for this too - it's not a simple typo (or something trivial) that doesn't need it - I'll create one unless @michaelandrepearce has already created one.

astitcher commented:
@michaelandrepearce can you paste in the JIRA report urls here then so they can be connected.

michaelandrepearce commented:
@astitcher https://issues.apache.org/jira/browse/PROTON-1572 just created with the output of cppcheck 1.8 run we did earlier today.

jdanekrh commented:
@michaelandrepearce I think there are two factors that reduced the number of issues found. I used CppCheck 1.79 and I invoked it from my IDE, which by default does not turn on all the switches. Furthermore, I ignored warnings from parts of SWIG code and anything I was not familiar with.

My goal was not fix everything. I just switched to new IDE (kdevelop), saw option to run cppcheck, so I hit it and just fixed few things I knew how.

I am now installing 1.80. I'll see if I can get the same output you have.

astitcher commented:
I've merged this now (with an extra fix)

jdanekrh commented:
@michaelandrepearce I went through some of the 1566 messages I got from CppCheck 1.80 (by category, I did not look into each in detail) and it seems to me that most of ones in the Error category are probably valid, except `Uninitialized struct member: bytes.{start,size}` (and I did not look into perlPERL_wrap.c). Many in the Warning category seem to be valid, too, except when it does not understand the SCNi64 macros. That is about 50 issues in total. The Info level messages are worth thinking about too, but probably not right now.

So, all in all, the tool works pretty well, I'd say. And while I am not a big fan of KDevelop, the integration with CppCheck is simple, but very hepful (clicking a message jumps me to the relevant code).

We should IMO consider checking in a suppression file for CppCheck, which would allow ignoring existing nonserious issues, and make new issues stand out more in the output.

michaelandrepearce commented:
@jdanekrh cppcheck is a command line tool in fact this is how I ran it, you don't need to use an ide plugin if you don't want to. Also there are Jenkins plugin to add to a build pipeline. Lastly as a note there's a plugin for visual studio, clion, eclipse ide's also if you're not a fan of kdevelop but prefer not to use cli tooling.
